ui/src/products-listing-module/products-listing-page/products-listing-page.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, HostListener, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { Categories } from 'src/Shared/models/Categories';
+import { Products } from 'src/Shared/models/Products';
+import { HomeServiceService } from 'src/Shared/services/home-service.service';
+import { ProductsListingServiceService } from 'src/Shared/services/products-listing-service.service';
+
+@Component({
+  selector: 'app-products-listing-page',
+  templateUrl: './products-listing-page.component.html',
+  styleUrls: ['./products-listing-page.component.sass']
+})
+export class ProductsListingPageComponent implements OnInit {
+
+  constructor(private productSListingService: ProductsListingServiceService, private router: Router, private homeService: HomeServiceService) {
+  }
+  categoriesData: Categories[] = [];
+  productsList: Products[] = [];
+  productsListCopy: Products[] = [];
+  currWinSize: any = window.innerWidth;
+  isDesktop: boolean = this.currWinSize > 1024;
+  isMobile: boolean = this.currWinSize < 768;
+  currCategory: any = null;
+  ngOnInit(): void {
+    if (sessionStorage.getItem('isLoggedIn') && sessionStorage.getItem('isLoggedIn') == 'true') {
+      this.homeService.categoryId.subscribe((response) => {
+        this.currCategory = response;
+        console.log(this.currCategory);
+        
+         this.filterProducts(this.currCategory);
+      });
+      this.getData();
+    }
+    else
+      this.router.navigate(['login']);
+  }
+  getData() {
+    this.productSListingService.getCategories().subscribe((response) => {
+      this.categoriesData = response;
+    }, (errorResponse) => {
+      console.log(errorResponse);
+    });
+    this.productSListingService.getProducts().subscribe((response) => {
+      this.productsList = response;
+      this.productsListCopy=response;
+    }, (errorResponse) => {
+      console.log(errorResponse);
+    });
+  }
+  @HostListener('window:resize', ['$event'])
+  onResize(event: any) {
+    this.isDesktop = event.target.innerWidth > 1024;
```

Comment:
or observe window size, to render different HTML conditionally on different viewports.

Use the same HTML and do this responsively

Reply:
Okay